### ISSUE TITLE
Renaming testClient

### ISSUE THREAD

**Issue by Yusuf007R** [honojs/hono#3192]
### What is the feature you are proposing?

I don't think the `testClient` should be called `testClient` because it's not only useful for testing. It's also very useful as a Server Side Caller. Let's say you're using Next.js and you're using Hono to manage your APIs. If, for some reason, in a SSR component you want to use `hc` to get some data, it would make a request to localhost to get the data from Hono (I'm not aware if Next.js does some kind of optimization to get the data without making the request).

However, I think there should be a way to do that with Hono, and that way is `testClient`. I feel like it makes more sense for `hc` to work like [Elysia Eden](https://elysiajs.com/eden/treaty/config.html#urlorinstance). This way, you would be able to send either a `URL` or a Hono instance to `hc`. If you send a Hono instance, it would function like the `testClient`.

**Comment by yusukebe** [comment]
Hi @Yusuf007R !

This is an interesting opinion.

>  If, for some reason, in a SSR component you want to use `hc` to get some data, it would make a request to localhost to get the data from Hono

I didn't think these cases well. Keep this issue, and let's consider it together. Thanks.

**Comment by Yusuf007R** [comment]
I could try implement it, I'm not really familiar with the codebase but I can try and get back to you if i get any problems

**Comment by yusukebe** [comment]
Good. Or, perhaps we have to share the ideal API design for it.

**Comment by Yusuf007R** [comment]
yeah that's kind of what I mean doing a Proof of concept

I feel like the ideal API would be something like this

```ts

const app = new Hono().get("/hello", (c) => c.json({ hello: "world" }));

const client = hc(app);
const res = await client.hello.$get();

// data = { hello: 'world' }
const data = await res.json();
```

also in case of implementing it, what do we do with `testClient`? do we deprecate it and recommend the use of `hc`?

**Comment by yusukebe** [comment]
@Yusuf007R 

The big point between the current `hc`and that your `hc` is that the latter is using the actual value `app` instead of just a TypeScrip Type.

```ts
// The current `hc`
// AppType is just a `type`
const client = hc<AppType>('http://localhost/')
```

Now, I've never investigated, so I'm not sure what problem it will bring, but passing actual value `app` may add additional code to client-side JavaScript when it was bundled.

**Comment by Yusuf007R** [comment]
Oh, it seems like I didn't explain myself properly.
the idea is being able to do both. So you can either pass a URL STRING or a hono instance. 
so you would never pass a hono instance in the client. just for SSR or testing. 
in the client you would use it the same way it does now

**Comment by Yusuf007R** [comment]
hello, @yusukebe what do you think about coming back to this? 🥺

**Comment by EdamAme-x** [comment]
I think it should be integrated because it is easy to integrate and easier to maintain.

**Comment by yusukebe** [comment]
@Yusuf007R 

I think it would be better to use `hc` and `testClient` as before rather than to have the `hc` code determine whether a URL string or a `hono` instance is used and branch the process. The roles are easier to understand and the code is cleaner.

**Comment by Yusuf007R** [comment]
Yes, that was actually my initial request. To rename testClient. Because it is really useful for SSR when using honojs on metaframeworks such as NEXTJS

**Comment by Yusuf007R** [comment]
could we revisit this topic? 😊 i really think it would be useful.

**Comment by yusukebe** [comment]
I don't want to do it. Changing the name of a helper that has already been created as a Testing Helper is a lot more work than you might think. If there are a lot of people who want this, I'll think about it.
